classes/02class/exercises/c02-network07/feernandobraga/README.md

```diff
@@ -0,0 +1,167 @@
+    response:
+    {
+        "AssociationId": "rtbassoc-05f98f635a8348a13",
+        "AssociationState": {
+            "State": "associated"
+        }
+    }
+
+    aws ec2 associate-route-table --route-table-id rtb-0e6c480035df5d6a7 --subnet-id subnet-065dc1da939f981f6
+    response:
+    {
+        "AssociationId": "rtbassoc-091f53415b8f2eff7",
+        "AssociationState": {
+            "State": "associated"
+        }
+    }
+
+    aws ec2 associate-route-table --route-table-id rtb-0e6c480035df5d6a7 --subnet-id subnet-009508e37b719dd77
+    response:
+    {
+        "AssociationId": "rtbassoc-08bbf333e59cf6273",
+        "AssociationState": {
+            "State": "associated"
+        }
+    }
+    ```
+
+- Commands for creating the following routes:
+
+| route table              | destination | target              |
+| ------------------------ | ----------- | ------------------- |
+| devopsacademy-rt-public  | 0.0.0.0     | `devopsacademy-igw` |
+| devopsacademy-rt-private | 0.0.0.0     | `devopsacademy-ngw` |
+
+```
+aws ec2 create-route --route-table-id rtb-0e6c480035df5d6a7 --destination-cidr-block 0.0.0.0/0 --gateway-id igw-06df7843381ed4767
+response:
+{
+    "Return": true
+}
+
+aws ec2 create-route --route-table-id rtb-0e0fae42163550983 --destination-cidr-block 0.0.0.0/0 --gateway-id nat-0b08ff466280436f1
+response:
+An error occurred (InvalidGatewayID.NotFound) when calling the CreateRoute operation: The gateway ID 'nat-0b08ff466280436f1' does not exist
+```
+
+- Answer the following questions:
+
+  - How did you accomplish both instructions?
+
+    ```
+    I was only able to accomplish the first one (public RT -> Internet gateway), since the NAT gateway was meant to be deleted right after exercise #06
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+
+    ```
+    This is the default route to any external interface. If the router can't find the destination IP address within the internal network, then it forwards the packet to this default address
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+
+    ```
+    This is the default route to any external interface. If the router can't find the destination IP address within the internal network, then it forwards the packet to this default address
+    ```
+
+  - What is the difference of IGW and NGW?
+
+    ```
+    An internet gateway (IGW) is what allows external traffic to/from a VPC.
+    A NAT gateway (NGW) allows for instances within a subnet to access the internet, but prevents connections to be initiated from outside the network.
+    ```
+
+  - Can you delete the destination route to your VPC network? Why?
+
+    ```
+    AWS only allows you to delete it if they are detached from the VPC, otherwise you can't. The reason is that the VPCs need to know how/where to route the traffic between instances and/or external resources.
```

Comment:
AWS does not allow:
- deleting the default route with destination `YOUR_VPC_CIDR_BLOCK` and gateway-id `local`.
- detaching the route table from a VPC

Please fix your answer.

Please fix this one.